content/blog/2025-04-02-introducing-bloom-filters.md

```diff
@@ -0,0 +1,188 @@
++++
+title= "Introducing Bloom Filters for Valkey"
+description = "Learn how to use bloom filters to perform large-scale membership testing with significant memory savings."
+date= 2025-04-02 01:01:01
+authors= [ "karthiksubbarao"]
++++
+
+We are introducing Bloom Filters as a new data type in Valkey via [valkey-bloom](https://github.com/valkey-io/valkey-bloom/), an official Valkey Module which is compatible with Valkey versions >= 8.0. Bloom filters provide efficient, large-scale membership testing, improving performance and offering significant memory savings for high-volume applications.
+
+As an example, previously, Valkey users could use the `SET` data type for handling advertisement deduplication workloads. However, since every item in the set is allocated, the problem with this approach is high memory usage. This article demonstrates how using the bloom filter data type from valkey-bloom achieves >93% memory savings for the same workload, while exploring its implementation, technical details, and practical recommendations.
+
+## Introduction
+
+Bloom filters are a space efficient probabilistic data structure that allows adding elements and checking whether elements exist. False positives are possible where a filter incorrectly indicates that an element exists, even though it was not added. However, Bloom Filters guarantee that false negatives do not occur, meaning it can never be the case that an element was added successfully and a filter reports it as not existing. Bloom Filters were first introduced in a paper from 1970 by Burton H. Bloom.
+
+<img src="/assets/media/pictures/bloomfilter_bitvector.png" style="width: 75%;" alt="Bloom Filter Bit Vector">
```

Comment:
Should be in markdown, we don't want to directly embed the html tags.

Reply:
Updated